pkg/router/kubernetes_default_test.go

```diff
@@ -281,9 +282,10 @@ func TestServiceRouter_Finalize(t *testing.T) {
 		},
 		Spec: corev1.ServiceSpec{
 			Ports: []corev1.ServicePort{{
-				Name:     "http",
-				Protocol: "TCP",
-				Port:     9898,
+				AppProtocol: &appProtocol,
```

Comment:
This test should be moved to `TestServiceRouter_Create`, since we want to test whether the services created by Flagger have the `AppProtocol` field set.

Reply:
🤦 apologies, good point. Misunderstood where this setting got applied, have updated the PR. Thanks for the feedback!